### PR TITLE
Harmonize widget return values

### PR DESCRIPTION
**Pull request by bdvorachek**
# Critical Changes

# Changes
- Updated the internal code that powers the way widgets communicate the values they contain when the form is saved. This should have no effect on existing functionality.
# Issues Closed

# Community Ideas Delivered

# New Metadata
- LWC
  - geConstants
# Deleted Metadata

### CODE REVIEW

**Comment by bdvorachek** [comment]
**lurch: attach W-040071

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HtPFQA0" target="_blank">W-040071</a>

**Comment by bdvorachek** [comment]
We'll want to merge #5315 before this, and that will introduce a few merge conflicts related to error handling. All that should be necessary there is to re-hook the error handler function in a new spot.

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 1]
```diff
@@ -882,9 +883,39 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
         }
 
-        // Set DataImport__c.Additional_Object_JSON__c if we've retrieved values from form widgets.
+        // resolve widget data to wherever it needs to go, a field on the di object or elsewhere.
         if (widgetValues) {
-            diRecord[ADDITIONAL_OBJECT_JSON__C] = JSON.stringify(widgetValues);
+            let diFieldPayloads = [];
+
+            widgetValues.forEach(widgetValue => {
+                if(widgetValue.type === 'diFieldValue' && !isUndefined(widgetValue.payload)) {
```
Comment (psadie-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:891]: Should we turn `diFieldValue` into a constant?

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 2]
```diff
@@ -882,9 +883,39 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
+            });
+
+            // diFieldValues is an array of objects where the key is the field name
+            // and the value is the value to be stored
+            let diFieldValuesArray;
+            try {
+                diFieldValuesArray = await Promise.all(diFieldPayloads);
+            } catch (ex) {
+                // ex will be whatever was sent to the reject() function in the original promise
+                // handle payment error here: handlePaymentError(ex);
```
Comment (psadie-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:903]:
```suggestion
                // TODO: handle payment error here: handlePaymentError(ex);
```

File: src/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js [hunk 3]
```diff
@@ -89,18 +91,43 @@ export default class geFormWidgetTokenizeCard extends LightningElement {
     */
     requestToken() {
         const iframe = this.template.querySelector(`[data-id='${this.CUSTOM_LABELS.commonPaymentServices}']`);
-        fireEvent(null, 'tokenRequested');
+
+        const tokenPromise = new Promise((resolve, reject) => {
+
+            const timer = setTimeout(() => {
+                reject('Request timed out');
```
[the remark this answers is not in the document]
Comment (bdvorachek) [reply src/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js:98]: I'm getting in touch with docs on this.

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 4]
```diff
@@ -883,9 +884,39 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
+            });
+
+            // diFieldValues is an array of objects where the key is the field name
+            // and the value is the value to be stored
+            let diFieldValuesArray;
+            try {
+                diFieldValuesArray = await Promise.all(diFieldPayloads);
+            } catch (ex) {
+                // ex will be whatever was sent to the reject() function in the original promise
+                // TODO: handle payment error here: handlePaymentError(ex);
```
Comment (ptewson-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:904]: I'm this hang on save when I have a card error (like expiration date in the past) - Is that related to this not being implemented? Console error shows "Uncaught (in promise) TypeError: Cannot read property 'forEach' of undefined
    at ie.buildDataImportRecord (geFormRenderer.js:4)"
